app/src/main/java/de/ur/explure/views/CategoryQueryFragment.kt

```diff
@@ -1,6 +1,82 @@
 package de.ur.explure.views
 
+import android.os.Bundle
+import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.crazylegend.viewbinding.viewBinding
 import de.ur.explure.R
+import de.ur.explure.adapter.SearchListAdapter
+import de.ur.explure.databinding.FragmentCategoryQueryBinding
+import de.ur.explure.viewmodel.CategoryViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class CategoryQueryFragment : Fragment(R.layout.fragment_category_query)
+class CategoryQueryFragment : Fragment(R.layout.fragment_category_query) {
+
+    private val binding by viewBinding(FragmentCategoryQueryBinding::bind)
+    private val args: CategoryQueryFragmentArgs by navArgs()
+
+    private val viewModel: CategoryViewModel by viewModel()
+    private lateinit var categoryAdapter: SearchListAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setTitle()
+        initializeAdapter()
+        observeRouteResult()
+        observeRouteModel()
+    }
+
+    private fun observeRouteModel() {
+        viewModel.categoryRoutes.observe(viewLifecycleOwner, { routes ->
+            categoryAdapter.submitList(routes)
+            binding.progressBar.visibility = View.GONE
+        })
+    }
+
+    private fun observeRouteResult() {
+        viewModel.noRoutes.observe(viewLifecycleOwner, {
+            if (it == true) {
+                binding.progressBar.visibility = View.GONE
+                binding.noResults.visibility = View.VISIBLE
+            }
+        })
+    }
+
+    private fun initializeAdapter() {
+        // viewModel.setupAlgolia()
+        val category = args.categoryQueryKey
+        categoryAdapter = SearchListAdapter { }
+
+        binding.recyclerViewCategoryResults.apply {
+            layoutManager = LinearLayoutManager(requireContext())
+            adapter = categoryAdapter
+        }
+
+        viewModel.getCategoryRoutes(category)
+    }
+
+    private fun setTitle() {
```

Comment:
Eine Datenbank ist extra dafür da um keine Daten in der App zu speichern, um diese dann zu erweitern/zu ändern. IDs und namen könnten sich z.B. ändern. 
Statt die ganzen Ids und Namen in der App zu speicher, übergib am besten das gesamte Category Object an dieses Fragment mit den safe args statt nur der category id. Dann hast du alle Daten (Id, Name, Resource, Color) auf dem Fragment und kannst diese nutzen.